arborist/server.go

```diff
@@ -334,20 +334,31 @@ func (server *Server) handleAuthProxy(w http.ResponseWriter, r *http.Request) {
 	authRequest.stmts = server.stmts
 	w.Header().Set("REMOTE_USER", authRequest.Username)
 
-	rv, err := authorizeUser(authRequest)
-	if err != nil {
-		msg := fmt.Sprintf("could not authorize user: %s", err.Error())
-		server.logger.Info("tried to handle auth request but input was invalid: %s", msg)
-		response := newErrorResponse(msg, 400, nil)
-		_ = response.write(w, r)
+	if (authRequest.Username == "") && (authRequest.ClientID == "") {
+		msg := "unauthorized: did not provide a username and/or client ID in request"
+		_ = newErrorResponse(msg, 403, nil).write(w, r)
 		return
 	}
-	if rv.Auth {
-		server.logger.Debug("user is authorized")
-	} else {
-		server.logger.Debug("user is unauthorized")
+
+	rv := &AuthResponse{}
+	rv.Auth = true
+	var err error = nil
```

Comment:
i don't think we need to initialize the err variable here

Reply:
i need to initialize it because of this logic:
```
if authRequest.Username != "" {
  rv, err = authorizeUser(authRequest)
}
if rv.Auth && authRequest.ClientID != "" {
  rv, err = authorizeClient(authRequest)
}
```
it needs to be initialized before i use `err =`. I could use `err :=` if i'm sure it's not initialized yet, but when i auth the client i don't know if i went through the user auth block or not

i don't work with Go a lot, there might be a better way of handling this 😬